src/AppInstallerCLICore/Resources.cpp

```diff
@@ -16,9 +15,25 @@ namespace AppInstaller::CLI::Resource
         return instance;
     }
 
-    Loader::Loader()
+    Loader::Loader() : m_wingetLoader(nullptr)
     {
-        m_wingetLoader = winrt::Windows::ApplicationModel::Resources::ResourceLoader::GetForViewIndependentUse(L"winget");
+        try
+        {
+            // The default constructor of ResourceLoader throws a winrt::hresult_error exception
+            // when resource.pri is not found. ResourceLoader::GetForViewIndependentUse also throws
+            // a winrt::hresult_error but for reasons unknown it only gets catch when running on the
+            // debugger. Running without a debugger will result in a crash that not even adding a
+            // catch all fix. To provide a good error message we call the default constructor
+            // before calling GetForViewIndependentUse.
+            m_wingetLoader = winrt::Windows::ApplicationModel::Resources::ResourceLoader();
+            m_wingetLoader = winrt::Windows::ApplicationModel::Resources::ResourceLoader::GetForViewIndependentUse(L"winget");
+        }
+        catch (const winrt::hresult_error& hre)
+        {
+            // This message cannot be localized.
+            AICLI_LOG(CLI, Error, << "Failure loading resource file with error: " << hre.code());
+            throw MissingResourceFileException();

```

Comment:
I would suggest actually making it `ResourceOpenException` or similar and pass the `hre.code()` in to it to create a message that better indicates the actual error.  It might be corrupt or access_denied or something else.  Maybe:
```
Could not open the resource file: 0x8007002: File not found.
```
Where we call the appropriate API to get the system error message for the second string part.